core-api/src/main/java/com/optimizely/ab/odp/ODPManager.java

```diff
@@ -0,0 +1,61 @@
+/**
+ *
+ *    Copyright 2022, Optimizely
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.odp;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class ODPManager {
+    private volatile ODPConfig odpConfig;
+    private final ODPSegmentManager segmentManager;
+    private final ODPEventManager eventManager;
+
+    public ODPManager(@Nonnull ODPConfig odpConfig, @Nonnull ODPApiManager apiManager) {
+        this(odpConfig, new ODPSegmentManager(odpConfig, apiManager), new ODPEventManager(odpConfig, apiManager));
+    }
+
+    public ODPManager(@Nonnull ODPConfig odpConfig, @Nonnull ODPSegmentManager segmentManager, @Nonnull ODPEventManager eventManager) {
+        this.odpConfig = odpConfig;
+        this.segmentManager = segmentManager;
+        this.eventManager = eventManager;
+        this.eventManager.start();
+    }
+
+    public ODPSegmentManager getSegmentManager() {
+        return segmentManager;
+    }
+
+    public ODPEventManager getEventManager() {
+        return eventManager;
+    }
+
+    public Boolean updateSettings(String apiHost, String apiKey, List<String> allSegments) {
+        ODPConfig newConfig = new ODPConfig(apiKey, apiHost, allSegments);
```

Comment:
Can we flush events in queue before updating apiKey/apiHost? It should not be common but when clients change the ODP account, all events in the queue should be flushed to the old account.